src/omv/imlib/jpeg.c

```diff
@@ -1173,20 +1223,30 @@ static int jpeg_processDU(jpeg_buf_t *jpeg_buf, int8_t *CDU, float *fdtbl, int D
         }
     }
 
+    jpeg_check_highwater(jpeg_buf); // check if we're getting close to the end of the buffer
+    // Use local vars to speed up buffer access
+    // and a macro (STORECODE) to manipulate the local vars
+    uint8_t *pOut, iBitCount; // output pointer and bit count
+    uint32_t ulBits; // accumulated bits
+    pOut = &jpeg_buf->buf[jpeg_buf->idx];
```

Comment:
@bitbank2 Assuming the above call in line 1226 to `jpeg_check_highwater` overflows and sets the `jpeg_buf->length` to 0,  pOut is set using the idx not length, and then the following code will overflow right ?

Perhaps just instead of resetting the length or idx, simply set the overflow flag and return true ?  For example:

```C
static bool jpeg_check_highwater(jpeg_buf_t *jpeg_buf)
{   
    if ((jpeg_buf->idx+1) >= jpeg_buf->length - 256) {
        if (jpeg_buf->realloc == false) {
            // Can't realloc buffer
            jpeg_buf->overflow = true;
        } else {
            jpeg_buf->length += 1024;
           jpeg_buf->buf = xrealloc(jpeg_buf->buf, jpeg_buf->length);
       }
    }
    return jpeg_buf->overflow;
} /* jpeg_check_highwater() */
```

Then:

```C
if (jpeg_check_highwater(jpeg_buf) == true) {
    return;
}
```

Reply:
Thanks for catching that mistake. I cleaned up the function to behave correctly.